compiler/luci-interpreter/src/kernels/Utils.cpp

```diff
@@ -116,6 +118,37 @@ void calculateActivationRangeQuantized(Activation activation, const Tensor *outp
                                         activation_max);
 }
 
+void computeMinMaxActivationInplace(Activation activation, float* data, int32_t size){
+  float activation_min, activation_max;
+  calculateActivationRange(activation, &activation_min, &activation_max);
+
+  for(int32_t i = 0; i < size; ++i){
+    data[i] = tflite::ActivationFunctionWithMinMax(data[i], activation_min, activation_max);
+  }
+}
+
+void computeActivationInplace(Activation activation, float *data, const int32_t size){
+  switch (activation)
+  {
+    case Activation::RELU:
+      computeMinMaxActivationInplace(activation, data, size);
+      break;
+    case Activation::RELU_N1_TO_1:
+      computeMinMaxActivationInplace(activation, data, size);
+      break;
+    case Activation::RELU6:
+      computeMinMaxActivationInplace(activation, data, size);
+      break;
+    case Activation::TANH:
+      tflite::reference_ops::Tanh({size}, data, {size}, data);
+      break;
+    case Activation::NONE:
+      break;
+    default:
+      throw std::runtime_error("Unsupported activation.");
+  }
+}
```

Comment:
It's needed to compute fused activation (include Tanh) for circle layers with custom kernels.